### PR TITLE
set email explicitly for v0.5

### PR DESCRIPTION
**Pull request by kunmingg**
For v0.5 request we need to get email from signed in identity.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4089)

### CODE REVIEW

**Comment by prodonjs** [comment]
Since we already try to retrieve the `email` value from the component's state, what if we just call `setState({email})` after line [443](https://github.com/kubeflow/kubeflow/blob/1198c95614ceeb1324bc1688281b1647a8dd90df/components/gcp-click-to-deploy/src/DeployForm.tsx#L443) since it's always going to be called before the request is built and sent.

**Comment by abhi-g** [comment]
Please address Jason's comment above

**Comment by kunmingg** [comment]
@prodonjs 
updated, thanks

File: components/gcp-click-to-deploy/src/DeployForm.tsx [hunk 1]
```diff
@@ -441,6 +441,9 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
         }
 
         const email = await Gapi.getSignedInEmail() || '';
+        this.setState({
+            ['email']: email,
```
Comment (avdaredevil) [inline components/gcp-click-to-deploy/src/DeployForm.tsx]: Kuming, you could write this whole blurb as:

```suggestion
            this.setState({email})
```

Comment (avdaredevil) [reply components/gcp-click-to-deploy/src/DeployForm.tsx]: Reference: https://alligator.io/js/object-property-shorthand-es6/

**Pull request merged**